utils/test_results.py

```diff
@@ -0,0 +1,74 @@
+import datetime as dt
+
+from django.contrib.postgres.aggregates import ArrayAgg
+from django.db.models import Avg, Case, FloatField, Value, When
+from shared.django_apps.reports.models import Test, TestInstance
+
+thirty_days_ago = dt.datetime.now(dt.UTC) - dt.timedelta(days=30)
+
+
+def aggregate_test_results(
+    repoid: int, branch: str | None = None, history: dt.timedelta | None = None
+):
```

Comment:
[nit] I'd love to see docstrings for this function (although reading through the code it's decently clear what it does. Good job with var naming